bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -238,28 +240,42 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
 
   @Override
   public ResultScanner<Row> readRows(ReadRowsRequest request) {
-    return readRows(request, retryOptions.enableRetries());
+    return readRows(request, retryOptions.enableRetries(), false);
+  }
+
+  @Override
+  public ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable) {
+    return readRows(request, retryOptions.enableRetries(), resumable);
   }
 
   /**
-   * Begin reading rows, optionally with a resumable scanner.
+   * Begin reading rows, optionally with a retriable or resumable scanner.
    */
-  private ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable) {
-    // Delegate all resumable operations to the scanner. It will request a non-resumable
-    // scanner during operation.
-    if (resumable) {
-      return new ResumingStreamingResultScanner(
-        retryOptions,
-          request,
-          new BigtableResultScannerFactory() {
+  private ResultScanner<Row> readRows(
```

Comment:
I'm not a fan of this method.  It did 2 things instead of a single thing, and now it does 3.  I'd rather this be broken up into three methods that each add their own